### ISSUE TITLE
Rename table on cluster does not change the path in zookeeper

### ISSUE THREAD

**Issue by VSMironov** [ClickHouse/ClickHouse#47228]
Hi!

I have unexpected zookeeper behavior when renaming a table in a cluster.
Creating a table
```
create table if not exists adc_by_geo_unit on cluster '{cluster}' (code Nullable(String), batch_id Int64) 
 engine = ReplicatedMergeTree order by (code) settings allow_nullable_key = 1;
```
Checking zookeeper

```
select * from system.zookeeper where path = 'clickhouse/tables/1/default' order by 1;

+---------------+-----+---------------------------+
|name           |value|path                       |
+---------------+-----+---------------------------+
|adc_by_geo_unit|     |clickhouse/tables/1/default|
+---------------+-----+---------------------------+
```

Rename the table and checking zookeeper

```
rename table adc_by_geo_unit to adc_by_geo_unit_rename on cluster '{cluster}';
select * from system.zookeeper where path = 'clickhouse/tables/1/default' order by 1;

+---------------+-----+---------------------------+
|name           |value|path                       |
+---------------+-----+---------------------------+
|adc_by_geo_unit|     |clickhouse/tables/1/default|
+---------------+-----+---------------------------+
```
The path has not changed, the table remains with the same name in zookepeer.
Checked if the table was renamed
```
select * from adc_by_geo_unit
Code: 60. DB::Exception: Table default.adc_by_geo_unit doesn't exist. (UNKNOWN_TABLE) (version 23.1.3.5 (official build))

select * from adc_by_geo_unit_rename
Ok
```

If we try to recreate the original table, we will get an error message.

```
create table if not exists adc_by_geo_unit on cluster '{cluster}' (code Nullable(String), batch_id Int64) 
 engine = ReplicatedMergeTree order by (code) settings allow_nullable_key = 1;

Code: 253. DB::Exception: There was an error on [127.0.0.1:9000]: 
Code: 253. DB::Exception: Replica /clickhouse/tables/1/default/adc_by_geo_unit/replicas/1 already exists. 
(REPLICA_ALREADY_EXISTS) (version 23.1.3.5 (official build)). (REPLICA_ALREADY_EXISTS) (version 23.1.3.5 (official build))
```

<details><summary>Server settings files</summary>
<p>

#### cluster.xml

```xml
<?xml version="1.0"?>
<yandex>
    <keeper_server>
        <tcp_port>9181</tcp_port>
        <server_id>1</server_id>
        <log_storage_path>/var/lib/clickhouse/coordination/log</log_storage_path>
        <snapshot_storage_path>/var/lib/clickhouse/coordination/snapshots</snapshot_storage_path>

        <coordination_settings>
            <operation_timeout_ms>10000</operation_timeout_ms>
            <session_timeout_ms>30000</session_timeout_ms>
            <raft_logs_level>warning</raft_logs_level>
        </coordination_settings>

        <raft_configuration>
            <server>
                <id>1</id>
                <hostname>127.0.0.1</hostname>
                <port>9444</port>
            </server>
        </raft_configuration>
    </keeper_server>

    <zookeeper>
        <node>
            <host>127.0.0.1</host>
            <port>9181</port>
        </node>
    </zookeeper>

    <remote_servers>
        <cluster>
            <secret>*******</secret>
            <shard>
                <internal_replication>true</internal_replication>
                <replica>
                    <host>127.0.0.1</host>
                    <port>9000</port>
                </replica>
            </shard>
        </cluster>
    </remote_servers>
</yandex>
```

#### config.xml

```xml
<?xml version="1.0"?>
<yandex>
        <listen_host replace="replace">0.0.0.0</listen_host>
        <default_profile>default</default_profile>
        <default_replica_path>/clickhouse/tables/{shard}/{database}/{table}</default_replica_path>
        <default_replica_name>{replica}</default_replica_name>

        <database_atomic_delay_before_drop_table_sec>1</database_atomic_delay_before_drop_table_sec>
        <distributed_ddl_allow_replicated_alter>1</distributed_ddl_allow_replicated_alter>
        <keep_alive_timeout>120</keep_alive_timeout>
        <max_server_memory_usage>64424509440</max_server_memory_usage>
        <shutdown_wait_unfinished>10</shutdown_wait_unfinished>
        <shutdown_wait_unfinished_queries>1</shutdown_wait_unfinished_queries>
</yandex>
```

#### macros.xml

```xml
<yandex>
    <macros>
        <cluster>cluster</cluster>
        <shard>1</shard>
        <replica>1</replica>
    </macros>
</yandex>
```

</p>
</details>

**Comment by den-crane** [comment]
By design.

The current approach to this problem is Atomic database engine (and Atomic is default) and UUID as ZK path.

create database A on cluster segmented;

create table A.A on cluster 'segmented'  ( A Int64 ) Engine=ReplicatedMergeTree order by A;

show create table A.A;

CREATE TABLE A.A
(
    `A` Int64
)
ENGINE = ReplicatedMergeTree('/clickhouse/tables/{uuid}/{shard}', '{replica}')
ORDER BY A
SETTINGS index_granularity = 8192;


show create table A.A settings show_table_uuid_in_table_create_query_if_not_nil=1;

CREATE TABLE A.A UUID '30d92172-0f4a-4fcf-93cb-1445c44c2260'
(
    `A` Int64
)
ENGINE = ReplicatedMergeTree('/clickhouse/tables/{uuid}/{shard}', '{replica}')
ORDER BY A
SETTINGS index_granularity = 8192;
```

**Comment by sbalasa** [comment]
---

For a similar issue I encountered with one of my tables, I applied the following resolution:

1. Manually instantiate the table on all nodes, excluding the `ON CLUSTER` clause during table creation.
2. If the error "REPLICA_ALREADY_EXISTS" is thrown, modify the ZooKeeper path slightly by appending or altering a segment of the path to ensure uniqueness.
3. Once the table is successfully created across nodes, execute the command:

`system sync replica on cluster '{cluster}' <db_name>.<table_name> strict ;`

This command ensures synchronization across replicas and should rectify the "REPLICA_ALREADY_EXISTS" error.
